src/util/props.js

```diff
@@ -0,0 +1,46 @@
+
+import { warn } from './warn'
+
+export function resolveProps (route, component, config) {
+  switch (typeof config) {
+
+    case 'undefined':
+      return
+
+    case 'object':
+      return config
+
+    case 'function':
+      return config(route)
+
+    case 'boolean':
+      if (!config) {
+        return
+      }
+      if (!component.props) {
+        return
+      }
+      const props = {}
+      for (const prop in route.params) {
+        const value = route.params[prop]
+        if (hasProp(component, prop)) {
+          const propType = component.props[prop].type
```

Comment:
I've simplified the behavior of the {props: true} which now sets the props to route.params. This was a problem in earlier versions, but as the params-to-props is now opt-in.